src/scf/newton.jl

```diff
@@ -0,0 +1,420 @@
+# Newton's algorithm for the direct minimization of the energy
+# ------------------------------------------------------------
+#
+# Newton algorithm consist of iterating over density matrices like
+#       P = P - δP
+# where δP solves
+#       (Ω+K)δP = [P,[P,H(P)]] (residual)
+# where (Ω+K) if the Jacobian of the direct minimization iterations. It is
+# defined as a super operator from the space tangent to the constraints
+# manifold at point P_∞, the solution of our problem.
+#   - K is the Hessian of the energy on the manifold;
+#   - Ω represents the influence of the curvature of the manifold :
+#         ΩδP = -[P_∞,[H(P_∞),δP]].
+#     In practice, we dont have access to P_∞ so we just use the current P.
+#     Ω is also -χ0^(-1) where χ0 is the occupation function.
```

Comment:
Sorry, my mistake, Ω = -(χ0)^{-1} where χ0 is the susceptibility (well I think so, but @antoine-levitt will confirm I guess). What do you mean by occupation function ?

Reply:
Okay, to be more precise, if you define the following operators

- E : builds a density matrix Eρ given a density ρ via Eρ(r,r') = δ(r,r')ρ(r)
- R : builds a density ρ given a density matrix O via ρ(r) = O(r,r)

then you have the relation χ0 = -R(Ω^-1)E

Reply:
Yes that's right. Michael I think you're right too (up to sign) except in this case it's a bit weird to define it like this because f is a step function

Reply:
What is the occupation function ? Which difference with χ0 then ?

Reply:
occupation function f(eps) = 1 if eps < eps_f, 0 otherwise. In the finite temperature case it becomes the Fermi-Dirac distribution. If you see this function as mapping matrices to matrices, its Frechet differential is -Omega^-1. Chi0 is as you wrote: the "diagonal" of -Omega^-1. It's confusing to write chi0^-1 though.

Reply:
Ok, I understand better, thanks. Maybe write χ0 = -R(Ω^-1)E into the code then to avoid writing χ0^-1 ?